source/grid/grid_tools.cc

```diff
@@ -3098,68 +3072,35 @@ namespace GridTools
                 Tensor<1,spacedim> h;
                 h[d] = eps/2;
 
-                if (respect_manifold == false)
-                  gradient[d]
-                    = ((objective_function (object, object_mid_point + h)
-                        -
-                        objective_function (object, object_mid_point - h))
-                       /
-                       eps);
-                else
-                  gradient[d]
-                    = ((objective_function (object,
-                                            manifold->project_to_surface(object,
-                                                                         object_mid_point + h))
-                        -
-                        objective_function (object,
-                                            manifold->project_to_surface(object,
-                                                                         object_mid_point - h)))
-                       /
-                       eps);
+                gradient[d] = (objective_function (object,
+                                                   project_to_object(object, object_mid_point + h))
```

Comment:
This seems like the wrong change. Before, it actually used the manifold to project, but now it just uses a Q1 mapping. Or do I misinterpret?

Reply:
It looks like I never commented on this during the initial revision. `project_to_object` uses manifold information to find a point on the object near the given point.